pkg/transformer/openshift/openshift.go

```diff
@@ -154,6 +284,36 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
 			objects = append(objects, o.initImageStream(name, service))
 		}
 
+		if service.Build != "" {
+			if !hasBuild {
```

Comment:
Okej ;-) I see it now.
 You don't want detect branch, remote and compose dir for every service with build in one docker-compose file. Because they will be the same.

I get it now. Thank you.